### PR TITLE
added /api/v3 check in internal/comment/github.go

### PR DESCRIPTION
**Pull request by varshneydevansh**
Validate --github-api-url #1980

We use GitHub Enterprise and provided an endpoint that included the “/api/v3/” suffix which Infracost appends on its own. The logging is inconsistent between these functions, and it should probably attempt to detect whether the suffix already exists before appending it.

### CODE REVIEW

**Review by varshneydevansh — COMMENTED** [review]
github.go file is updated correctly with the check for the "/api/v3" in the URL. This file change belongs to the issue #2534 

However, in [kubernetes_cluster_node_pool.go](https://github.com/infracost/infracost/pull/2534/files#diff-4a93ab291f5dba270b6b6633bab1425b5936d16b844a0b61b77c77aaaca82d4e) file there could be some edge cases where the `regex` might not work, but the probability of this is low, and also **I haven't run the test cases yet** for this file. 
This change belongs to the issue Infracost uses Premium disk for AKS even though it's not available for given VM family #2407

File: internal/comment/github.go [hunk 1]
```diff
@@ -113,6 +113,10 @@ func newGitHubAPIClients(ctx context.Context, token string, apiURL string, tlsCo
 	if !strings.HasSuffix(u.Path, "/") {
 		u.Path += "/"
 	}
+	// Check if "/api/v3/" already exists in the URL path.
+	if strings.Contains(u.Path, "/api/v3/") {
```
Comment (aliscott) [inline internal/comment/github.go:117]:
```suggestion
	if strings.HasSuffix(u.Path, "/api/v3/") {
```
We could use `HasSuffix` here to make sure we are only checking the end of the string.

Comment (varshneydevansh) [reply internal/comment/github.go:117]:
```go
	// Check if "/api/v3/" already exists in the URL path.
	if strings.HasSuffix(u.Path, "/api/v3/") {
		return nil, nil, fmt.Errorf("the /api/v3/ suffix should not be included in the --github-api-url")
	}
```
After this, we also have to lowercase the first **The** as it will show an error **error string should not be capitalized**
